lib/src/model/common/node.dart

```diff
@@ -192,9 +192,14 @@ abstract class Node {
     bool prepend = false,
   }) {
     final pos = nodeAt(path).position;
-    final (newPos, newSan) = pos.makeSan(convertAltCastlingMove(move) ?? move);
+    final isKingMove =
+        pos.board.roleAt(parseSquare(move.uci.substring(0, 2)) ?? 0) ==
```

Comment:
You could use `move.from` instead of `move.uci`.

Reply:
Ok. The from property is only available on instances of NormalMove, so I check if its a NormalMove first. Castling will always be a normalmove anyways.